### PR TITLE
[DOCS] Add prerequisites for CCS

### LINKED ISSUE #72001: Cross cluster search doc to specify the requirement on the local cluster for a node with remote_cluster_client role
Just for reference, CCR [doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/ccr-getting-started.html) mentions

> In the local cluster, all nodes with the master node role must also have the remote_cluster_client role. The local cluster must also have at least one node with both a data role and the remote_cluster_client role. Individual tasks for coordinating replication scale based on the number of data nodes with the remote_cluster_client role in the local cluster. 

CCS [doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/modules-cross-cluster-search.html#ccs-remote-cluster-setup) does not seem to mention this requirement so I think it would make sense to add something like this : 

> In the local cluster, ensures at least one node has remote_cluster_client [role](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/modules-node.html#node-roles) 

For reference, if all nodes do not have that role, search on any remote index will not query the remote cluster and return `IndexNotFoundException`

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes

* Adds a prerequisites section covering remote cluster config, node roles, and security.
* Moves existing content about remote cluster config to the prereqs.
* Updates the remote cluster docs to include information about eligible gateway nodes and tagging for gateway nodes.

Closes https://github.com/elastic/elasticsearch/issues/72001

### Previews
* CCS prereqs: https://elasticsearch_81782.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-prereqs
* Sniff mode: https://elasticsearch_81782.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/remote-clusters.html#sniff-mode

### CODE REVIEW

File: docs/reference/search/search-your-data/search-across-clusters.asciidoc [hunk 1]
```diff
@@ -22,6 +20,37 @@ The following APIs support {ccs}:
...
+[discrete]
+[[ccs-prereqs]]
+=== Prerequisites
+
+* {ccs-cap} requires remote clusters. To set up remote clusters on {ess},
+see link:{cloud}/ec-enable-ccs.html[configure remote clusters on {ess}]. If you
+run {es} on your own hardware, see <<remote-clusters>>.
++
+To ensure your remote cluster configuration supports {ccs}, see
+<<ccs-supported-configurations>>.
```
Comment (jrodewig) [inline docs/reference/search/search-your-data/search-across-clusters.asciidoc:32]: We're updating the section for this xref in https://github.com/elastic/elasticsearch/pull/80809.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @jrodewig. Sorry for the late review.

**Review by javanna — APPROVED** [review]

**Comment by jrodewig** [comment]
Thanks @dnhatn @javanna!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------:|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/82367)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/82368)|
|✅|7.16|[](https://github.com/elastic/elasticsearch/pull/82369)|

**Pull request merged**